icechunk/tests/test_s3_storage.rs

```diff
@@ -0,0 +1,190 @@
+use std::{collections::HashSet, sync::Arc};
+
```

Comment:
There is some rebase weirdness in here. I think I have renamed this file to test_storage, and added significantly to it